src/phoenix/db/insertion/span.py

```diff
@@ -28,42 +28,80 @@ async def insert_span(
     dialect = SupportedSQLDialect(session.bind.dialect.name)
     if (
         project_rowid := await session.scalar(
-            select(models.Project.id).where(models.Project.name == project_name)
+            select(models.Project.id).filter_by(name=project_name)
         )
     ) is None:
         project_rowid = await session.scalar(
-            insert(models.Project).values(dict(name=project_name)).returning(models.Project.id)
+            insert(models.Project).values(name=project_name).returning(models.Project.id)
         )
     assert project_rowid is not None
-    if trace := await session.scalar(
-        select(models.Trace).where(models.Trace.trace_id == span.context.trace_id)
-    ):
+
+    trace_session: Optional[models.TraceSession] = None
+    session_id = get_attribute_value(span.attributes, SpanAttributes.SESSION_ID)
+    if session_id:
+        session_id = str(session_id)
+        assert isinstance(session_id, str)
+        trace_session = await session.scalar(
+            select(models.TraceSession).filter_by(session_id=session_id)
+        )
+        if not trace_session:
+            trace_session = await session.scalar(
+                insert(models.TraceSession)
+                .values(
+                    project_id=project_rowid,
+                    session_id=session_id,
+                    start_time=span.start_time,
+                    end_time=span.end_time,
+                )
+                .returning(models.TraceSession)
+            )
```

Comment:
Is it possible to just perform an upsert?

Reply:
yes, i can change

Reply:
Going to punt on this because it turns out to be not so straightforward given the conditional statement regarding the `project_id`. It is in fact quite doable, but I would have to write the upsert query from scratch (for each backend) and separately from our existing helper function for upsert. The same applies to upsert of traces.